libnymea-mqtt/mqttclient.cpp

```diff
@@ -364,9 +364,9 @@ void MqttClientPrivate::onReadyRead()
             }
             socket->write(retryPacket.serialize());
         }
+        restartKeepAliveTimer();
```

Comment:
is there a reason for this change?

Reply:
Ah, just saw the full commit message. ok. that makes sense to me. Thanks. Perhaps it could also make sense to adjust an existing auto-test to check for this more correct behavior.

Reply:
as written in the commit message:

> By chaning the order, MqttClient::isConnected() returns `true`
in slots connected to MqttClient::connected().

Reply:
I cannot find any test that can be easily extended. Any recommendation?

Reply:
I've added one. Made a pull request against your master. I don't mind if you want to merge mine into yours first or we merge both into upstream master.